### ISSUE TITLE
add export genesis tests

### ISSUE THREAD

**Issue by colin-axner** [cosmos/ibc-go#243]
## Summary

we need tests which ensure that all the genesis fields are exported and that the exported genesis is valid

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged/assigned

**Comment by fedekunze** [comment]
have you considered adding in-store migrations instead?

**Comment by colin-axner** [comment]
> have you considered using the new in-store migrations instead?

yup! Already implemented that in #205. Unfortunately we should keep genesis upgrades around as a backup solution. As it stands, it isn't possible to change the chain id just via an x/upgrade since tendermint loads the chain id from the gen doc and doesn't expose a way to change it block to block

Supporting genesis upgrades is a lot of work and the process is clumsy at best, but if for some reason something goes wrong with x/upgrade or a chain id must be incremented, it is nice knowing there's always a backup option available
